src/script/lua_api/l_nodemeta.cpp

```diff
@@ -64,8 +64,9 @@ void NodeMetaRef::reportMetadataChange(const std::string *name)
 	SANITY_CHECK(!m_is_local);
 	// NOTE: This same code is in rollback_interface.cpp
 	// Inform other things that the metadata has changed
-	NodeMetadata *meta = dynamic_cast<NodeMetadata*>(m_meta);
-
+	NodeMetadata *meta = dynamic_cast<NodeMetadata*>(m_meta ? m_meta : getmeta(true));
```

Comment:
Hack: I observed that m_meta appears to pretty much alway be NULL (unless it's local meta, which it never is in a server environment?), so I'm getting it here. Should probably be obtained somewhere else.